crates/nu-protocol/src/value/mod.rs

```diff
@@ -1813,6 +1786,42 @@ impl Value {
         Ok(())
     }
 
+    /// Creates a new [Value] with the specified member at the specified path.
+    /// This is used by [Value::insert_data_at_cell_path] and [Value::upsert_data_at_cell_path] whenever they have the need to insert a non-existent element
+    pub fn with_data_at_cell_path(
```

Comment:
```suggestion
    fn with_data_at_cell_path(
```

This can be private I think.